doc/api/diagnostics_channel.md

```diff
@@ -407,6 +455,507 @@ channel.subscribe(onMessage);
+  store.getStore(); // Span({ some: 'message' })
+});
+```
+
+### Class: `TracingChannel`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+The class `TracingChannel` is a collection of [TracingChannel Channels][] which
+together express a trace. It is used to formalize and simplify the process of
+producing tracing events. [`diagnostics_channel.tracingChannel()`][] is used
+to construct a `TracingChannel`. As with `Channel` it is recommended to create
+and reuse a single `TracingChannel` at the top-level of the file rather than
+creating them dynamically.
+
+#### `tracingChannel.subscribe(subscribers)`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+* `subscribers` {Object} Set of [TracingChannel Channels][] subscribers
+  * `start` {Function} The [`start` event][] subscriber
+  * `end` {Function} The [`end` event][] subscriber
+  * `asyncStart` {Function} The [`asyncStart` event][] subscriber
+  * `asyncEnd` {Function} The [`asyncEnd` event][] subscriber
+  * `error` {Function} The [`error` event][] subscriber
+
+Helper to subscribe a collection of functions to the corresponding channels.
+This is the same as calling [`channel.subscribe(onMessage)`][] on each channel
+individually.
+
+```mjs
+import diagnostics_channel from 'node:diagnostics_channel';
+
+const channels = diagnostics_channel.tracingChannel('my-channel');
+
+channels.subscribe({
+  start(message) {
+    // Handle start message
+  },
+  end(message) {
+    // Handle end message
+  },
+  asyncStart(message) {
+    // Handle asyncEnd message
+  },
+  asyncEnd(message) {
+    // Handle asyncEnd message
+  },
+  error(message) {
+    // Handle error message
+  },
+});
+```
+
+```cjs
+const diagnostics_channel = require('node:diagnostics_channel');
+
+const channels = diagnostics_channel.tracingChannel('my-channel');
+
+channels.subscribe({
+  start(message) {
+    // Handle start message
+  },
+  end(message) {
+    // Handle end message
+  },
+  asyncStart(message) {
+    // Handle asyncEnd message
+  },
+  asyncEnd(message) {
+    // Handle asyncEnd message
+  },
+  error(message) {
+    // Handle error message
+  },
+});
+```
+
+#### `tracingChannel.unsubscribe(subscribers)`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+* `subscribers` {Object} Set of [TracingChannel Channels][] subscribers
+  * `start` {Function} The [`start` event][] subscriber
+  * `end` {Function} The [`end` event][] subscriber
+  * `asyncStart` {Function} The [`asyncStart` event][] subscriber
+  * `asyncEnd` {Function} The [`asyncEnd` event][] subscriber
+  * `error` {Function} The [`error` event][] subscriber
+* Returns: {boolean} If all handlers were successfully unsubscribed
+
+Helper to unsubscribe a collection of functions from the corresponding channels.
+This is the same as calling [`channel.unsubscribe(onMessage)`][] on each channel
+individually.
+
+```mjs
+import diagnostics_channel from 'node:diagnostics_channel';
+
+const channels = diagnostics_channel.tracingChannel('my-channel');
+
+channels.unsubscribe({
```

Comment:
My understanding is that references to the original functions used to subscribe are required. With this example code in its current form it might not be as obvious to the reader that references to the original functions are required.